src/components/PeopleList.js

```diff
@@ -1,50 +1,38 @@
 import { GoHeart, GoHeartFill } from 'react-icons/go';
-import { useState, useEffect } from 'react';
 
-function PeopleList({ starWarsChars }) {
-    const [toggleFavourite, setToggleFavourite] = useState({});
-
-    useEffect(() => {
-        const savedFavourite = localStorage.getItem("toggleFavourite");
-        if (savedFavourite) {
-            setToggleFavourite(JSON.parse(savedFavourite));
-        }
-    }, []);
-
-
-    const handleToggleFavourite = (id) => {
-        setToggleFavourite((toggleStatus) => {
-            const savedStatus = {
-                ...toggleStatus,
-                [id]: !toggleStatus[id]
-            };
-            localStorage.setItem("toggleFavourite", JSON.stringify(savedStatus));
-            return savedStatus;
-        });
-    };
-
-
-
-    return (
-        <div>
-            <h3 className="p-5 text-xl lg:text-4xl text-yellow-500 font-bold flex justify-center">Star Wars Characters</h3>
-            <div className="mx-10 text-yellow-300 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5">{starWarsChars.map((starWarsChar, id) => {
```

Comment:
Ebben a függvényben az id a ciklus, a loop indexe. Tehát ha mondjuk az ötödik karakternél járunk, akkor az id itt 4, hiszen 0-tól indulunk.

Alapvetően nem egy rossz ötlet ezt használni arra, hogy eltároljuk kedvencnek, viszont, ha mondjuk továbbmegyünk és egy következő oldalnyi karaktert is fetch-elünk az API-tól, akkor már nem feltétlen fog működni.

Vagy ha eltároljuk mondjuk a 2-t, mert a harmadik karaktert beszivecskéztem, viszont az API megváltozik vagy az első oldal helyett a második oldalt kérjük le, akkor azt fogja mutatni, hogy a harmadik karaktert beszivecskéztem, de igazából nem is azt szivecskéztem.

Úgyhogy emiatt én végül is a karakter nevét tárolom el, másik fájlban láthatod!